pos_disable_pricelist_selection/static/src/js/SetPricelistButton.js

```diff
@@ -0,0 +1,37 @@
+odoo.define("pos_disable_pricelist_selection.SetPricelistButton", function (require) {
+    "use strict";
+
+    const SetPricelistButton = require("point_of_sale.SetPricelistButton");
+    const Registries = require("point_of_sale.Registries");
+
+    const SetPricelistButtonSelect = (SetPricelistButton) =>
+        class extends SetPricelistButton {
+            async onClick() {
+                // Replaced to use selectable_pricelists instead of pricelists
+                const selectionList = this.env.pos.selectable_pricelists.map(
+                    (pricelist) => ({
+                        id: pricelist.id,
+                        label: pricelist.name,
+                        isSelected: pricelist.id === this.currentOrder.pricelist.id,
+                        item: pricelist,
+                    })
+                );
+
+                const {confirmed, payload: selectedPricelist} = await this.showPopup(
+                    "SelectionPopup",
+                    {
+                        title: this.env._t("Select the pricelist"),
+                        list: selectionList,
+                    }
+                );
+
+                if (confirmed) {
+                    this.currentOrder.set_pricelist(selectedPricelist);
+                }
+            }
+        };
```

Comment:
**question (non-blocking)**: I don't know if this could be a better approach, but I wonder if, instead of overwriting the entire method, we could quietly replace `this.env.pos.pricelists` with `this.env.pos.selectable_pricelists`, then call super(), and finally put things back where they belong. My main concern is improving compatibility with other modules. What do you think?

Reply:
agree